akka-stream/src/main/scala/akka/stream/scaladsl/SourceWithContext.scala

```diff
@@ -9,6 +9,17 @@ import scala.annotation.unchecked.uncheckedVariance
 import akka.annotation.ApiMayChange
 import akka.stream._
 
+/**
+ * API MAY CHANGE
+ */
+@ApiMayChange
+object SourceWithContext {
+  /**
+   * Creates a SourceWithContext from a regular source that operates on a pair of `(data, context)` elements.
+   */
+  def from[CO, O, M](source: Source[(O, CO), M]): SourceWithContext[CO, O, M] = new SourceWithContext(source)
```

Comment:
why is this named `from` and in javadsl it is `fromPairs`?

would be more future friendly to name it `fromPairs` because it would not be possible to add another overloaded `from(source: Source[SomeOtherType, M])` since they would have same type after erasure

Reply:
I followed the pattern from `FlowWithSource`, but you are right it can be more explicit.
I'll check if we can come up with better names.